pkg/kubectl/cmd/apply.go

```diff
@@ -245,7 +245,13 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	// Create the versioned struct from the type defined in the restmapping
 	// (which is the API version we'll be submitting the patch to)
 	versionedObject, err := api.Scheme.New(p.mapping.GroupVersionKind)
-	if err != nil {
+	switch {
+	case runtime.IsNotRegisteredError(err):
+		versionedObject, _, err = p.decoder.Decode(current, nil, nil)
```

Comment:
> generic patch computation

Do you mean `JSONPatchType` or `MergePatchType`?

> complete replace mode

I guess you mean option 2 in  https://github.com/kubernetes/community/issues/229#issuecomment-271636721